stdlib/public/core/StringInterpolation.swift

```diff
@@ -26,48 +101,127 @@ extension String : _ExpressibleByStringInterpolation {
+  /// interpolation to create a new string by including values, literals,
+  /// variables, or expressions enclosed in parentheses, prefixed by a
+  /// backslash (`\(`...`)`).
   ///
-  /// Do not call this initializer directly. It is used by the compiler when
-  /// interpreting string interpolations.
+  ///     let price = 2
+  ///     let number = 3
+  ///     let message = "If one cookie costs \(price) dollars, " +
+  ///                   "\(number) cookies cost \(price * number) dollars."
+  ///     print(message)
+  ///     // Prints "If one cookie costs 2 dollars, 3 cookies cost 6 dollars."
   @inlinable
-  public init<T>(stringInterpolationSegment expr: T) {
-    self = String(describing: expr)
+  public mutating func appendInterpolation<T: TextOutputStreamable>(_ value: T) {
+    value.write(to: &_storage)
   }
-
-  /// Creates a string containing the given value's textual representation.
+  
+  /// Interpolates the given value's textual representation into the
+  /// string literal being created.
+  /// 
+  /// Do not call this method directly. It is used by the compiler when
+  /// interpreting string interpolations. Instead, use string
+  /// interpolation to create a new string by including values, literals,
+  /// variables, or expressions enclosed in parentheses, prefixed by a
+  /// backslash (`\(`...`)`).
   ///
-  /// Do not call this initializer directly. It is used by the compiler when
-  /// interpreting string interpolations.
+  ///     let price = 2
+  ///     let number = 3
+  ///     let message = "If one cookie costs \(price) dollars, " +
+  ///                   "\(number) cookies cost \(price * number) dollars."
+  ///     print(message)
+  ///     // Prints "If one cookie costs 2 dollars, 3 cookies cost 6 dollars."
   @inlinable
-  public init<T: TextOutputStreamable> (stringInterpolationSegment expr: T) {
-    self = _toStringReadOnlyStreamable(expr)
+  public mutating func appendInterpolation<T: CustomStringConvertible>(_ value: T) {
+    _storage += value.description
   }
-
-  /// Creates a string containing the given value's textual representation.
+  
+  /// Interpolates the given value's textual representation into the
+  /// string literal being created.
+  /// 
+  /// Do not call this method directly. It is used by the compiler when
+  /// interpreting string interpolations. Instead, use string
+  /// interpolation to create a new string by including values, literals,
+  /// variables, or expressions enclosed in parentheses, prefixed by a
+  /// backslash (`\(`...`)`).
   ///
-  /// Do not call this initializer directly. It is used by the compiler when
-  /// interpreting string interpolations.
+  ///     let price = 2
+  ///     let number = 3
+  ///     let message = "If one cookie costs \(price) dollars, " +
+  ///                   "\(number) cookies cost \(price * number) dollars."
+  ///     print(message)
+  ///     // Prints "If one cookie costs 2 dollars, 3 cookies cost 6 dollars."
+  @inlinable
```

Comment:
Why is this `@inlinable`?

Reply:
In general, these benchmarked faster when they were inlinable, and IIRC the `init(stringInterpolationSegment:)` overloads they replaces were inlinable/transparent. We can retest/reconsider this, though.